packages/editor/src/components/post-featured-image/index.js

```diff
@@ -31,7 +31,7 @@ function PostFeaturedImage( { currentPostId, featuredImageId, onUpdateImage, onR
 
 	let mediaWidth, mediaHeight, mediaSourceUrl;
 	if ( media ) {
-		const mediaSize = applyFilters( 'editor.PostFeaturedImage.imageSize', 'post-thumbnail', media.id, currentPostId );
+		const mediaSize = applyFilters( 'editor.PostFeaturedImage.imageSize', 'thumbnail', media.id, currentPostId );
```

Comment:
Hi @derweili, thank you for your contribution 👍
I verified that the classic editor also loads post thumbnail when it is available.
E.g: this is the markup classic editor provided (the src is for the post-thumbnail size):
<img width="493" alt="Screenshot 2019-05-27 at 22 19 23" src="https://user-images.githubusercontent.com/11271197/58439798-09f23f80-80ce-11e9-9dad-5c0f43ae59ce.png">

I verified that if post_thumbnail is not available thumbnail size used in the classic editor while in Gutenberg full size is used. I think maybe we should follow the same logic verify if post_thumbnail is available and if yes we should use it, if not Gutenberg verifies if thumbnail is available.

Reply:
Makes sense to me. The `post-thumbnail` image size is available if the theme adds support for post thumbnails.

`thumbnail` on the other hand is always there, but is usually something like 150x150 big.

Reply:
@swissspidy `thumbnail` is not always there. 

If the uploaded image is smaller than `thumbnail` size (defaut 150 x 150) the thumbnail size is missing.
`full` is the only size that's always available.

I think that the image sizes should be chosen in the following order:
1. Load `post-thumbnail` (still use the `editor.PostFeaturedImage.imageSize` filter)
2. If `post-thumbnail` is not available, load `thumbnail`
3. If both `post-thumbnail` and `thumbnail` are not available, use the `full` size.

Reply:
Hi @derweili, that seems a good plan 👍

Reply:
@jorgefilipecosta I implemented the changes as described.
I reset the `mediaSize` to `post-thumbnail` and added a new `fallbackMediaSize` which has the value `thumbnail` by default.

I also added a new filter `editor.PostFeaturedImage.fallbackImageSize` to modify the new `fallbackMediaSize`, similar to the existing filter `editor.PostFeaturedImage.imageSize`

Reply:
> `full` is the only size that's always available.

The frustrating thing here is that `media.media_details.sizes` is an empty object when the image is smaller than the `thumbnail` size. If it always contained a size then we could do something to pick the appropriate size in the order of preference. 

Given the current structure of the `media.media_details` object, this change looks good to me.

Alternatively, we could make another change so `sizes` always contains `full` at a minimum. 

Any thoughts?

Reply:
Thank you for the improvements @derweili this PR seems to be on a good path I left a suggestion.

> Alternatively, we could make another change so sizes always contains full at a minimum.

It would be good if originally that was the case but now it may be risky to change the structure as there may be external code/filters that is expecting and empty array.
I think the current direction addresses the problem well.